File: app/src/main/java/com/rockthevote/grommet/data/db/model/Session.java
package com.rockthevote.grommet.data.db.model;

import android.os.Parcelable;
import android.provider.BaseColumns;
import androidx.annotation.Nullable;

/**
 * Created by Mechanical Man, LLC on 7/14/17. Grommet
 */

public abstract class Session implements Parcelable, BaseColumns {

    public enum SessionStatus {
        NEW_SESSION("new"),
        PARTNER_UPDATE("partner_update"),
        SESSION_CLEARED("session_cleared"),
        DETAILS_ENTERED("details_entered"),
        CLOCKED_IN("clocked_in"),
        CLOCKED_OUT("clocked_out");

        private final String type;

        SessionStatus(String type) {
            this.type = type;
        }

        @Override
        public String toString() {
            return type;
        }

        @Nullable
        public static SessionStatus fromString(String type) {
            for (SessionStatus val : values()) {
                if (val.toString().equals(type)) {
                    return val;
                }
            }
            return null;
        }
    }

}

File: app/src/main/java/com/rockthevote/grommet/data/db/model/Session.kt
package com.rockthevote.grommet.data.db.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.rockthevote.grommet.data.api.model.ApiGeoLocation
import java.util.*

/**
 * Created by Mechanical Man on 4/12/20.
 */
@Entity(tableName = "session")
data class Session(
        @PrimaryKey
        @ColumnInfo(name = "session_id")
        val sessionId: Long = 0,

        @ColumnInfo(name = "partner_info_id")
        val partnerInfoId: Long,

        // some combination of canvasser name and epoch string
        @ColumnInfo(name = "source_tracking_id")
        val sourceTrackingId: String,

        // the canvasser zip code is the partner tracking id
        @ColumnInfo(name = "partner_tracking_id")
        val partnerTrackingId: String,

        @ColumnInfo(name = "geo_location") val geoLocation: ApiGeoLocation,

        // name of the canvasing event, the location field
        @ColumnInfo(name = "open_tracking_id")
        val openTrackingId: String,

        @ColumnInfo(name = "canvasser_name")
        val canvasserName: String,

        // the tablet number
        @ColumnInfo(name = "device_id")
        val deviceId: String,

        @ColumnInfo(name = "abandoned_count")
        val abandonedCount: Int = 0,

        @ColumnInfo(name = "registration_count")
        val registrationCount: Int = 0,

        /**
         * Number of registrants that opted into receiving SMS during a session
         */
        @ColumnInfo(name = "sms_count")
        val smsCount: Int = 0,

        /**
         * Number of registrants that provided DL info during a session
         */
        @ColumnInfo(name = "drivers_license_count")
        val driversLicenseCount: Int = 0,

        /**
         * Number of registrants that provided SSN info during a session
         */
        @ColumnInfo(name = "ssn_count")
        val ssnCount: Int = 0,

        /**
         * Number of registrants that opted in to receiving emails
         */
        @ColumnInfo(name = "email_count")
        val emailCount: Int = 0,

        @ColumnInfo(name = "clock_in_time")
        val clockInTime: Date? = null,

        @ColumnInfo(name = "clock_out_time")
        val clockOutTime: Date? = null,

        @ColumnInfo(name = "session_status")
        val sessionStatus: SessionStatus = SessionStatus.PARTNER_UPDATE
)